mmengine/hooks/sync_buffer_hook.py

```diff
@@ -13,6 +13,24 @@ class SyncBuffersHook(Hook):
 
     def __init__(self) -> None:
         self.distributed = is_distributed()
+        # A flag to mark whether synchronization has been done in
+        # after_train_epoch
+        self.called_in_train = False
```

Comment:
We could set `self.called_in_train=True` by default. Then the unnecessary buffer synchronization will not happen when we call `runner.val()` separately.

Reply:
Looks like such setting will make buffer sync before validation in IterBasedTrainLoop not work. Correct me if I am wrong.

Reply:
Ahh, you're right, just ignore this comment!